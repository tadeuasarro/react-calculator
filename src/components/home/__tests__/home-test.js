import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Home from '../home';

describe('Display', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><Home /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a Home component', () => {
    render(<BrowserRouter><Home /></BrowserRouter>);
    const element = screen.getByText('Welcome to Math-magicians!!!');
    expect(element).toBeInTheDocument();
  });
});
