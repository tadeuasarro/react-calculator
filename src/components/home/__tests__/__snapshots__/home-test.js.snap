// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<main>
  <div
    id="navbar"
  >
    <h1>
      Math-magicians
    </h1>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/app"
          onClick={[Function]}
        >
          App
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    id="outer-container"
  >
    <div
      id="home"
    >
      <h2>
        Welcome to Math-magicians!!!
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus euismod quis turpis viverra dictum. Suspendisse nibh justo, porta vitae lorem ac, rhoncus ultricies nibh. Quisque accumsan maximus libero, vel vulputate augue venenatis id. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce sapien nulla, ullamcorper eu eros vitae, auctor condimentum elit. Praesent sit amet magna massa. Nunc suscipit fermentum bibendum. Curabitur pulvinar, quam facilisis mattis mattis, risus metus ultricies ex, non porttitor nulla felis et nunc. Duis vel leo felis. Nulla volutpat luctus mi sed mollis.
      </p>
      <p>
        Donec tincidunt condimentum dolor, vel pellentesque magna volutpat eu. Morbi ultrices maximus quam, a lobortis tortor consectetur sit amet. Ut quis metus eleifend, fringilla dolor ut, luctus ante. Cras condimentum at dolor in venenatis. Aenean porta et lorem eget ultricies. Praesent non dolor eu libero sollicitudin rhoncus. Phasellus quis ante efficitur, malesuada massa id, feugiat erat. Nulla at nibh erat.
      </p>
    </div>
  </div>
</main>
`;
